scripts/generate_sitemap.py

```diff
@@ -0,0 +1,210 @@
+        """Writes and gzips each sitemap xml file"""
+        path = os.path.join(self.sitemap_dir, 'sitemap_{}.xml'.format(str(self.sitemap_count)))
+        print('Writing and gzipping `{}`: url_count = {}'.format(path, str(self.url_count)))
+
+        xml_str = self.doc.toprettyxml(indent="  ", encoding='utf-8')
+        with open(path, 'w') as f:
+            f.write(xml_str)
+        # Write zipped file
+        with open(path, 'rb') as f_in, gzip.open(path + '.gz', 'wb') as f_out:
+            shutil.copyfileobj(f_in, f_out)
+        self.sitemap_count += 1
+
+    def write_sitemap_index(self):
+        """Writes the index file for all of the sitemap files"""
+        doc = xml.dom.minidom.Document()
+        sitemap_index = self.doc.createElement('sitemapindex')
+        sitemap_index.setAttribute('xmlns', 'http://www.sitemaps.org/schemas/sitemap/0.9')
+        doc.appendChild(sitemap_index)
+
+        for f in range(self.sitemap_count):
+            sitemap = doc.createElement('sitemap')
+            sitemap_index.appendChild(sitemap)
+
+            loc = doc.createElement('loc')
+            sitemap.appendChild(loc)
+            loc_text = self.doc.createTextNode(urlparse.urljoin(settings.DOMAIN, 'sitemaps/sitemap_{}.xml.gz'.format(str(f))))
+            loc.appendChild(loc_text)
+
+            datemod = doc.createElement('datemod')
+            sitemap.appendChild(datemod)
+            datemod_text = self.doc.createTextNode(datetime.datetime.now().isoformat())
+            datemod.appendChild(datemod_text)
+
+        print('Writing `sitemap_index.xml`')
+        xml_str = doc.toprettyxml(indent="  ", encoding='utf-8')
+        with open(os.path.join(self.sitemap_dir, 'sitemap_index.xml'), 'w') as f:
+            f.write(xml_str)
+
+    def log_errors(self, obj, obj_id, error):
+        if not self.errors:
+            script_utils.add_file_logger(logger, __file__)
+        self.errors += 1
+        logger.info('Error on {}, {}:'.format(obj, obj_id))
+        logger.exception(error)
+        if self.errors == 1000:
+            sentry.log_message('ERROR: generate_sitemap stopped execution after reaching 1000 errors. See logs for details.')
+            raise Exception('Too many errors generating sitemap.')
+
+    def generate(self):
+        print('Generating Sitemap')
+
+        # Progress bar
+        progress = Progress()
+
+        # Preprint urls
+        objs = paginated(PreprintService, Q('is_published', 'eq', True))
+        # Can't do a proper percentage bar, as can't count total ahead
+        progress.start(100, 'PREP: ')
+        for obj in objs:
+            progress.increment()
+            if progress.count >= 100:
+                progress.restart()
+            if not obj.node or not obj.primary_file or obj.node.is_deleted or not obj.node.is_public:
+                continue
+            try:
+                preprint_date = obj.date_modified.isoformat() if obj.date_modified else obj.date_created.isoformat()
```

Comment:
@brianjgeiger The date thing I mentioned is handled here by using date_modified but defaulting to date_created. For this version I think that is fine. If it is a common issue on prod data, it is probably a bigger problem than just the sitemap.